packages/integration-tests/test/transferTest.js

```diff
@@ -221,4 +221,232 @@ describe('Transfer tests', async () => {
+
+      // Alice deposits ETH into the Plasma contract
+      await helper.depositEthAndWait(rootChain, childChain, aliceAccount.address, DEPOSIT_AMOUNT_ETH, aliceAccount.privateKey)
+      console.log(`Alice deposited ${DEPOSIT_AMOUNT_ETH} into RootChain contract`)
+
+      // Send ERC20 tokens to Alice's account
+      await helper.fundAccountERC20(web3, testErc20Contract, accounts[0], '', aliceAccount.address, INTIIAL_ALICE_AMOUNT_ERC20)
+      // Account must approve the Plasma contract
+      await helper.approveERC20(web3, testErc20Contract, aliceAccount.address, aliceAccount.privateKey, rootChain.plasmaContractAddress, DEPOSIT_AMOUNT_ERC20)
+      // Create the ERC20 deposit transaction
+      const depositTx = transaction.encodeDeposit(aliceAccount.address, DEPOSIT_AMOUNT_ERC20, config.testErc20Contract)
+      // Deposit ERC20 tokens into the Plasma contract
+      await rootChain.depositToken(depositTx, { from: aliceAccount.address, privateKey: aliceAccount.privateKey })
+      console.log(`Alice deposited ${DEPOSIT_AMOUNT_ERC20} tokens into RootChain contract`)
+
+      // Wait for transaction to be mined and reflected in the account's balance
+      const balance = await helper.waitForBalance(childChain, aliceAccount.address, DEPOSIT_AMOUNT_ERC20, ERC20_CURRENCY)
+      console.log(`Alice's balance: ${balance[0].amount.toString()}`)
+    })
+
+    it('should transfer ETH and ERC20 tokens on the childchain', async () => {
+      // Check utxos on the child chain
+      const utxos = await childChain.getUtxos(aliceAccount.address)
+      assert.equal(utxos.length, 2)
+      assert.hasAllKeys(utxos[0], ['utxo_pos', 'txindex', 'owner', 'oindex', 'currency', 'blknum', 'amount'])
+      assert.equal(utxos[0].amount, DEPOSIT_AMOUNT_ETH)
+      assert.equal(utxos[0].currency, transaction.ETH_CURRENCY)
+      assert.equal(utxos[1].amount, DEPOSIT_AMOUNT_ERC20)
+      assert.equal(utxos[1].currency, ERC20_CURRENCY)
+
+      const CHANGE_AMOUNT_ETH = utxos[0].amount - TRANSFER_AMOUNT_ETH
+      const CHANGE_AMOUNT_ERC20 = utxos[1].amount - TRANSFER_AMOUNT_ERC20
+      const txBody = {
+        inputs: [utxos[0], utxos[1]],
+        outputs: [{
+          owner: bobAccount.address,
+          currency: transaction.ETH_CURRENCY,
+          amount: Number(TRANSFER_AMOUNT_ETH)
+        }, {
+          owner: aliceAccount.address,
+          currency: transaction.ETH_CURRENCY,
+          amount: CHANGE_AMOUNT_ETH
+        }, {
+          owner: bobAccount.address,
+          currency: ERC20_CURRENCY,
+          amount: Number(TRANSFER_AMOUNT_ERC20)
+        }, {
+          owner: aliceAccount.address,
+          currency: ERC20_CURRENCY,
+          amount: CHANGE_AMOUNT_ERC20
+        }]
+      }
+
+      // Create the unsigned transaction
+      const unsignedTx = childChain.createTransaction(txBody)
+      // Sign it
+      const signatures = await childChain.signTransaction(unsignedTx, [aliceAccount.privateKey, aliceAccount.privateKey])
```

Comment:
why is `aliceAccount.privateKey` repeated twice in this array ? Is it because we needed two inputs for this test

Reply:
Yep. You need to sign each input, and there's no point in the code trying to figure out if the inputs belong to the same address - let the user do that.